studio-drm-standalone/src/receiver/receiver.js

```diff
@@ -8,14 +8,30 @@
     // When the cast player is loaded, fetch the drm token and license url and set up the playback config
     playerManager.setMessageInterceptor(cast.framework.messages.MessageType.LOAD, loadRequestData => {
         let customData = loadRequestData.media.customData;
-        token = customData["token"];
-        laurl = customData["laurl"];
+        // Allows mobile devices to send the source file in their own way
+        let source = loadRequestData.media.contentUrl || loadRequestData.media.entity || loadRequestData.media.contentId;
+        if (!source || source == ""){
+            let error = new cast.framework.messages.ErrorData(
+            cast.framework.messages.ErrorType.LOAD_FAILED
+            );
+            error.reason = cast.framework.messages.ErrorReason.INVALID_REQUEST;
+            return error;
+        }
+        if (customData){
+            if (customData["laurl"]){
```

Comment:
In the absence of DRM on content, the expected behaviour would be to playback successfully. I didn't want to force a token because that interferes with this behaviour

Reply:
Does it? that sounds weird to me, what is the `laurl` used for when there isn't any DRM?

Reply:
Nothing, the laurl and token are only set if there's custom data coming from the request. If there's no custom data, there's no DRM information available

Reply:
Yes. But that comes from the sender app, so by the point it would fail here, it would already fail in the sender app

Reply:
Are the sender app and the receiver app linked in such a way that out sender app has to use our receiver app? Could someone not write their own version of the sender app and use our receiver?

Reply:
The sender app is whatever is doing the casting - this repo has an example, but the JWP web player is a sender app as well.

What the custom data object looks like will change depending on the sender app, but these receivers work for native mobile (assuming `laurl` and `token` are named & configured appropriately on the sender side) and the JWP web player

Reply:
This "assuming laurl and token are named & configured appropriately on the sender side" is the key point, we shouldn't be assuming this, if we need there to be both the `laurl` and `token` in the custom data both should have validation to check they exist before we try to use them.

Reply:
Ahh, I understand! Thank you, amended - is this any better?